independent-projects/arc/runtime/src/main/java/io/quarkus/arc/NoClassInterceptors.java

```diff
@@ -0,0 +1,16 @@
+package io.quarkus.arc;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * If a business method is annotated {@code @NoClassInterceptors}, then class-level interceptors
+ * are ignored for this method. In other words, the only interceptors that apply to such business
+ * method are interceptors declared directly on the method.
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.METHOD)
```

Comment:
What about `@AroundConstruct`?

Reply:
I've ignored lifecycle callbacks completely, but to be fair, I don't know if the change here affects them or not :-) Let me check.

Reply:
OK so I changed this line to `@Target({ElementType.METHOD, ElementType.CONSTRUCTOR})` and the class-level interceptor still applies. That is, the change in this PR isn't enough to work for constructors. I think it would be a pretty simple change to `BeanInfo.initLifecycleInterceptors` to achieve that, but do we actually want it?

Reply:
I actually made the change right now and it works. So I can add that (with tests and docs adjustments of course).

Reply:
> That is, the change in this PR isn't enough to work for constructors.

Ofc, it's not ;-)

> I actually made the change right now and it works. So I can add that (with tests and docs adjustments of course).

+1

Reply:
Cool, I'll update this PR shortly (once I understand why a test I wrote fails :-) ).

Reply:
I almost feel ashamed to admit that it took me half an hour to figure out that an `@AroundConstruct` interceptor isn't invoked, because the bean under test is normal-scoped and hence instantiated lazily :laughing:

Updated.